pipeline.py

```diff
@@ -27,62 +36,132 @@ class MorphologicalToken:
+        i = 1
+        for filename in Path(self.path_to_raw_txt_data).iterdir():
+            if re.fullmatch(r'.+\d+_raw.txt', str(filename)):
+                storage[i] = Article(url=None, article_id=i)
+                storage[i].text = storage[i].get_raw_text()
+                i += 1
+        return storage
 
     def get_articles(self):
         """
         Returns storage params
         """
-        pass
+        return self._storage
 
 
 class TextProcessingPipeline:
     """
     Process articles from corpus manager
     """
-    def __init__(self, corpus_manager: CorpusManager):
-        pass
+
+    def __init__(self, corpus_manager):
+        self.corpus_manager = corpus_manager
+        self._current_article = None
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles()
+        for article in articles.values():
+            self._current_article = article
+            tokens = self._process()
+            tokens = ' '.join([str(token) for token in tokens])
+            article.save_processed(tokens)
 
     def _process(self) -> List[type(MorphologicalToken)]:
         """
         Performs processing of each text
         """
-        pass
+        morph = pymorphy2.MorphAnalyzer()
+
+        text = self._current_article.text
+        text = ' '.join(re.findall(r'\w+', text.lower()))
+        result = Mystem().analyze(text)
+        tokenized_text = []
+
+        for word in result[::2]:  # skip whitespaces
+            if 'analysis' not in word or word['analysis'] == []:
+                continue
+
+            original = word['text']
+            normalized = word['analysis'][0]['lex']
+            tags = word['analysis'][0]['gr']
+
+            token = MorphologicalToken(original, normalized)
+            token.mystem_tags = tags
+
+            if pymorpy_res := morph.parse(original):
+                token.pymorpy_tags = pymorpy_res[0].tag
+
+            tokenized_text.append(token)
+        return tokenized_text
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path_to_validate = Path(path_to_validate)
+
+    if not path_to_validate.exists():
+        raise FileNotFoundError
+
+    if not any(True for _ in path_to_validate.iterdir()):
+        raise EmptyDirectoryError
+
+    if not path_to_validate.is_dir():
+        raise NotADirectoryError
+
+    max_files_number = max(len(list(path_to_validate.glob('*.txt'))),
+                           len(list(path_to_validate.glob('*.json'))))
+    prev_not_exists = False
+    for i in range(1, max_files_number + 1):
+        raw_path = path_to_validate / f'{i}_raw.txt'
+        meta_path = path_to_validate / f'{i}_meta.json'
+
+        if raw_path.exists() and meta_path.exists() and not prev_not_exists:
+            continue
+        if not raw_path.exists() and not meta_path.exists():
+            prev_not_exists = True
```

Comment:
why do not we fail immediately?

Reply:
it could be the last of our files and next files are extra